File: typescript/tuple.ts
type StringNumberPair = [string, number];
let pair: StringNumberPair = ["hello", 100];
/*
console.log(pair[0]);
console.log(pair[1]);
*/
function div(a : number, b: number):[any,number]
{
    if(b === 0)
    {
        return ["Divide by zero",0];
    }
    else
    {
        return [null,a/b];
    }
}

let [err,result] = div(10,2);
console.log("Error ",err);
console.log("Result ",result);

File: typescript/tuple.go
package main

import (
	"errors"
	"fmt"
)

func div(a int, b int) (error, int) {
	if b == 0 {
		return errors.New("Divide by zero"), 0
	} else {
		return nil, a / b
	}

}
func main() {
	err, result := div(10, 0)

	if err == nil {
		fmt.Println(result)
	} else {
		fmt.Println("Error ", err)
	}

}
